packages/fast-check/src/arbitrary/infiniteStream.ts

```diff
@@ -5,15 +5,20 @@ import { StreamArbitrary } from './_internals/StreamArbitrary';
 /**
  * Produce an infinite stream of values
  *
+ * WARNING: By default, infiniteStream remembers all values it has ever
+ * generated. This causes unbounded memory growth during large tests.
+ * Set history=false to disable.
+ *
  * WARNING: Requires Object.assign
  *
  * @param arb - Arbitrary used to generate the values
+ * @param history - Whether to remember generated values (since 4.3.0)
  *
  * @remarks Since 1.8.0
  * @public
  */
-function infiniteStream<T>(arb: Arbitrary<T>): Arbitrary<Stream<T>> {
-  return new StreamArbitrary(arb);
+function infiniteStream<T>(arb: Arbitrary<T>, history: boolean = true): Arbitrary<Stream<T>> {
```

Comment:
Prefer: `constraints?: Contraints` with `Contraints = {noHistory?: boolean}`. It will be more aligned with our usual way to define constraints on other arbitraries.

```suggestion
function infiniteStream<T>(arb: Arbitrary<T>, constraints: InfiniteStreamConstraints = {}): Arbitrary<Stream<T>> {
```

Reply:
Latest commit creates `InfiniteStreamConstraints` with a single member `noHistory`. Constructor parameter is still the affirmative `history`.

Reply:
Perfect :+1: